subsystems/sidecar/lib/model.js

```diff
@@ -0,0 +1,89 @@
+'use strict'
+const HyperDB = require('hyperdb')
+const DBLock = require('db-lock')
+const hypercoreid = require('hypercore-id-encoding')
+const dbSpec = require('../../../spec/db')
+const { PLATFORM_HYPERDB } = require('../../../constants')
+
+module.exports = class Model {
+  constructor () {
+    this.db = HyperDB.rocks(PLATFORM_HYPERDB, dbSpec)
+
+    this.lock = new DBLock({
+      enter: () => {
+        return this.db.transaction()
+      },
+      exit: tx => {
+        return tx.flush()
+      },
+      maxParallel: 1
+    })
+  }
+
+  async getBundle (link) {
+    link = hypercoreid.isValid(link) ? hypercoreid.normalize(link) : link // if not valid, it is a path
```

Comment:
hypercoreid cannot tell you if something is a valid link. use our link module. the db is both for local and remote apps ie file and pear links

Reply:
Reverted/Unresolved, there's something broken that needs fixing first

@rafapaezbas I think this was a mistake to do in addBundle()

```js
link = hypercoreid.isValid(link) ? hypercoreid.normalize(link) : link
```

To clarify, printed all values of param getBundle `link` from our test case, got 3 types:

```
/Users/user/holepunch/pear/test

5hnz6pk1pohm63iu1r5yde1th4fq71ao4y8oimaxxt9s67ryueiy

Ɨ��~�s.s���.��0����%s������
```

the Buffer one shouldn't exist, right? @mafintosh pear-link relevant here? I thought we received a pear:// here but not really.

full log
```
getBundle link: /Users/user/holepunch/pear/test
getBundle link: �5Rl8�f��6�Q����
getBundle link: Ժ��XM	.*T��	�B����բ�|��2\
getBundle link: ��ks>0p῍,+�:�G{�6
getBundle link: 417x7dnajwr1hko4k1fqmmaj4edrfux9u1ftiipnoj6czsadgjqy
getBundle link: 5hnz6pk1pohm63iu1r5yde1th4fq71ao4y8oimaxxt9s67ryueiy
getBundle link: sz8gsrmu8aquy85ohg9a4mbmbhd3cqzndbdzzntsbkm4mgmzxseo
getBundle link: Ժ��XM	.*T��	�B����բ�|��2\
getBundle link: �5Rl8�f��6�Q����
getBundle link: ��ks>0p῍,+�:�G{�6
getBundle link: Ժ��XM	.*T��	�B����բ�|��2\
getBundle link: �5Rl8�f��6�Q����
getBundle link: Ɨ��~�s.s���.��0����%s������
getBundle link: a4moxfzsx5j8gmuud6mm36bqusjbgcg46zchs93fqquchghrsxeo
getBundle link: Ɨ��~�s.s���.��0����%s������
getBundle link: Ɨ��~�s.s���.��0����%s������
getBundle link: �O

                  ���3����6�:V�ݙ�q7���(�2
getBundle link: zi8osfemuky6qch1d8hb91g5g5buwiip5sc5ahjz1khxnkgqgefy
getBundle link: �O

                  ���3����6�:V�ݙ�q7���(�2
getBundle link: �O

                  ���3����6�:V�ݙ�q7���(�2
getBundle link: ����

getBundle link: wneczaombewck7n1s85q57hwhesqqmsdxqccp36dg8hc3rru4boy
getBundle link: ����

getBundle link: ����

getBundle link: r������`��fs�U1A����p�%���c
getBundle link: qmaezfpm94xgbchxc339hijterj57eed16gzyywcrz1m5z9bopto
getBundle link: r������`��fs�U1A����p�%���c
getBundle link: r������`��fs�U1A����p�%���c
getBundle link: ���u���?�1�%Eh�ʅ���;S�UϹ
getBundle link: dnphbdehqzt6dp9d86hddqbfewdgtg6kown6poxm8cyi8sni36ho
getBundle link: ���u���?�1�%Eh�ʅ���;S�UϹ
getBundle link: ���u���?�1�%Eh�ʅ���;S�UϹ
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker/
getBundle link: /Users/user/holepunch/pear/test/fixtures/print-args/
getBundle link: gc1�
                    ��`��0a0���B��.#��pw�A�@
getBundle link: c7tudfym78wgbigygboubdiyz7bmjybqrcyb37saqb5obqnbtiyy
getBundle link: gc1�
                    ��`��0a0���B��.#��pw�A�@
getBundle link: /Users/user/holepunch/pear/test/fixtures/hello-world/
getBundle link: V{樋��85�oН#�׍u7��i1i"���
getBundle link: k37op3ietcc5z7yt8yjdmm5x4nq186szti4ux6c6pras1ez7snby
getBundle link: 4� �կ���;p�)�!�S�VBz�6)H�
getBundle link: g141besii6ro9uqw8pamykczr8zf8yksej7moptjjdjyixx4xgfo
getBundle link: V{樋��85�oН#�׍u7��i1i"���
getBundle link: 4� �կ���;p�)�!�S�VBz�6)H�
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-parent/
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-end-from-child/
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-parent-error-handler/
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-end-from-parent/
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-destroy-from-child/
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-child/
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-destroy-from-parent/
getBundle link: ���2
                    �1Sp�-��(��o�МP��Pvm��
getBundle link: 45o9kcomhgodnw5oths4573eoxwg97goutebsae7zx5iy7upagko
getBundle link: /Users/user/holepunch/pear/test/fixtures/worker-child-error-handler/
getBundle link: ���2
                    �1Sp�-��(��o�МP��Pvm��
getBundle link: xq6bjk35aiss1oitweg8y79f4oaemy49q7ap1hhohisiyg6py13o
getBundle link: {��;�miB��
                          pw��0��_wp�s��mP���
getBundle link: bJ����Ų��o0Ō�U�"���B�	O�@
getBundle link: cjfe4yhfhax95tp14aqho53oasgqqictrk95tinn3hrw9keesiyy
getBundle link: bJ����Ų��o0Ō�U�"���B�	O�@
getBundle link: ��K$>�d��r�"�XI}#�ʈ�5��{�t
getBundle link: ��:�̞S�毜F70��"٢@BJ�P ၌��|{
getBundle link: bJ����Ų��o0Ō�U�"���B�	O�@
getBundle link: cjfe4yhfhax95tp14aqho53oasgqqictrk95tinn3hrw9keesiyy
getBundle link: bJ����Ų��o0Ō�U�"���B�	O�@
getBundle link: cjfe4yhfhax95tp14aqho53oasgqqictrk95tinn3hrw9keesiyy
getBundle link: ���I��]���p(%;��~#�Ɉ1��|�
getBundle link: 76mkjuzy7jrh37n7s9gp6hberwxyqq9g5f9n8aqjtyatzy5h7qiy
getBundle link: ���I��]���p(%;��~#�Ɉ1��|�
getBundle link: ?��+w��n��nH����w���e�~�v/�S
getBundle link: 866ob4bmq9kob6dqy5iph51enzt63o65q97afjuf9p9je7tx57jo
getBundle link: $��>�u�d���u��^�����ܘ�4l�$��<
getBundle link: ruq9rxsbqsegj7cnzf44mpk66u39ng7k51cyp63wpuknj3eqoc6y
getBundle link: ��Ʊ��3�n�j?h,��
getBundle link: ym3jihggs8zckchqyfzjn4t9pysefpok7gu1ygh7433faj17scjo
getBundle link: )�[% ��$���9���ӈ,���EwK>�
getBundle link: fgcfsjjysghbyjy5incu1gpes8yy7whef19amuodn3nzq1ad84no
getBundle link: )�[% ��$���9���ӈ,���EwK>�
```